src/workspaces/NewWorkspaceModal/NewWorkspaceWizard.tsx

```diff
@@ -0,0 +1,818 @@
+          name,
+          authorizationDomain: _.map((v) => ({ membersGroupName: v }), [...getRequiredGroups(), ...groups]),
+          attributes: { description },
+          copyFilesWithPrefix: isGoogleBillingProject() ? 'notebooks/' : 'analyses/',
+          ...(!!bucketLocation && isGoogleBillingProject() && { bucketLocation }),
+          enhancedBucketLogging,
+          addUsers: acl,
+        };
+
+        const createdWorkspace = await Utils.cond(
+          [
+            !!cloneWorkspace,
+            async () => {
+              const workspace: WorkspaceInfo = await Workspaces()
+                .workspaceV2(cloneWorkspace!.workspace.namespace, cloneWorkspace!.workspace.name)
+                .clone(body);
+              const featuredList = await FirecloudBucket().getFeaturedWorkspaces();
+              const metricsData = {
+                featured: _.some(
+                  { namespace: cloneWorkspace!.workspace.namespace, name: cloneWorkspace!.workspace.name },
+                  featuredList
+                ),
+                ...extractCrossWorkspaceDetails(cloneWorkspace!, { workspace }),
+                fromWorkspaceRegion: sourceGCPWorkspaceRegion,
+                toWorkspaceRegion: bucketLocation,
+              };
+              void Metrics().captureEvent(Events.workspaceClone, metricsData);
+              return workspace;
+            },
+          ],
+          async () => {
+            const workspace = await Workspaces().create(body);
+            const metricsData = {
+              ...extractWorkspaceDetails(workspace),
+              region: bucketLocation,
+            };
+            void Metrics().captureEvent(Events.workspaceCreate, metricsData);
+            return workspace;
+          }
+        );
+
+        onSuccess(createdWorkspace);
+      } catch (error: unknown) {
+        const errorMessage = await (async () => {
+          if (error instanceof Response) {
+            try {
+              const { message } = await error.json();
+              return message || 'Unknown error.';
+            } catch (readResponseError) {
+              return 'Unknown error.';
+            }
+          }
+          if (error instanceof Error) {
+            return error.message;
+          }
+          return 'Unknown error.';
+        })();
+        setCreating(false);
+        setCreateError(errorMessage);
+      }
+    };
+
+    const loadData = _.flow(
+      withErrorReportingInModal('Error loading data', onDismiss),
+      Utils.withBusyState(setLoading)
+    )(() =>
+      Promise.all([
+        Billing(signal)
+          .listProjects()
+          .then(_.filter({ status: 'Ready' }))
+          .then(_.filter((project: BillingProject) => isBillingProjectApplicable(project)))
+          .then((projects: BillingProject[]) => {
+            setBillingProjects(projects);
+            setNamespace(_.some({ projectName: namespace }, projects) ? namespace : undefined);
+          }),
```

Comment:
definitely out of scope for this PR: this Rawls API consistently takes 2+ seconds for me, which makes opening the modal slow. I wonder if this is due to Azure billing projects and once those are gone, this will be faster.